publish.gradle

```diff
@@ -13,23 +13,44 @@
 // limitations under the License.
 
 apply plugin: 'maven-publish'
+apply plugin: 'signing'
+
 afterEvaluate {
+
     publishing {
         repositories {
             maven {
+                name = 'mavenRepo'
                 url = findProperty('mavenRepo') ?: "${buildDir}/repo"
             }
+            maven {
+                def releasesRepoUrl = "https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/"
+                def snapshotsRepoUrl = "https://s01.oss.sonatype.org/content/repositories/snapshots/"
+
+                url = releaseVersion.endsWith('SNAPSHOT') ? snapshotsRepoUrl : releasesRepoUrl
```

Comment:
ℹ️ Adding SNAPSHOT to the end of the version will switch the publication repo URL to maven snapshot.